akka-typed/src/test/java/akka/typed/javadsl/ActorCompile.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+package akka.typed.javadsl;
+
+import akka.typed.*;
+import static akka.typed.javadsl.Actor.*;
+
+public class ActorCompile {
+  
+  interface MyMsg {}
+  
+  class MyMsgA implements MyMsg {
+    final ActorRef<String> replyTo;
+
+    public MyMsgA(ActorRef<String> replyTo) {
+      this.replyTo = replyTo;
+    }
+  }
+
+  class MyMsgB implements MyMsg {
+    final String greeting;
+
+    public MyMsgB(String greeting) {
+      this.greeting = greeting;
+    }
+  }
+
+  Behavior<MyMsg> actor1 = signalOrMessage((ctx, signal) -> same(),  (ctx, msg) -> stopped());
+  Behavior<MyMsg> actor2 = stateful((ctx, msg) -> unhandled());
+  Behavior<MyMsg> actor3 = stateless((ctx, msg) -> {});
+  Behavior<MyMsg> actor4 = empty();
+  Behavior<MyMsg> actor5 = ignore();
+  Behavior<MyMsg> actor6 = tap((ctx, signal) -> {}, (ctx, msg) -> {}, actor5);
+  Behavior<MyMsgA> actor7 = actor6.narrow();
+  Behavior<MyMsg> actor8 = deferred(ctx -> {
+    final ActorRef<MyMsg> self = ctx.getSelf();
+    return monitor(self, ignore());
+  });
+  Behavior<MyMsg> actor9 = widened(actor7, pf -> pf.match(MyMsgA.class, x -> x));
+
+  {
+    Actor.<MyMsg>stateful((ctx, msg) -> {
+      if (msg instanceof MyMsgA) {
```

Comment:
Here we just need the user to return the next behavior, it is not a partial function or similar. I am not sure we should try to improve upon the Java way of type matching, even though we find it loquacious. In this example it also does not get more concise than this version, but I admit that that is due to handling only a single type in each step.

Anyway, we can provide a pattern matching library independently of behavior—or more precisely: independent of Akka—at a later time.